midp/gfx.js

```diff
@@ -278,10 +268,27 @@ var currentlyFocusedTextEditor;
         converterFunc(rgbData, abgrData, width, height, 0, width);
 
         context.putImageData(ctxImageData, 0, 0);
-
-        setImageData(imageData, width, height, context);
     };
 
+    function setImageDataDimensions(imageData, w, h) {
+        imageData.klass.classInfo.getField("I.width.I").set(imageData, w);
+        imageData.context.canvas.width = w;
+
+        imageData.klass.classInfo.getField("I.height.I").set(imageData, h);
+        imageData.context.canvas.height = h;
+    }
+
+    Override["javax/microedition/lcdui/ImageData.<init>.()V"] = function() {
+        this.context = document.createElement("canvas").getContext("2d");
```

Comment:
I had done a `grep -r "new ImageData"` to see where we're constructing `ImageData` objects. One of our tests - "tests/midlets/favouriteartists/Control.java" - constructs its own `ImageData` object so it seemed like it might be legitimate to try to construct an `ImageData` outside of the `ImageDataFactory`. If that test is invalid, I can move construction of `ImageData` objects into the `ImageDataFactory` methods.